### ISSUE TITLE
Revisit Existing Rules

### ISSUE THREAD

**Issue by kahnjw** [yola/jsgreat#22]
I think some of the existing rules in jsgreat are likely a bit overly opinionated. We should revisit these rules, and try to formulate new, or alternative rules. 

My first target is: https://github.com/yola/jsgreat#anonymous-callback-functions

With the advent of ES6 arrows, we can concisely write one line functions, and get around this rule. There are no open and closing brackets in arrow functions to newline. That said there are going to be applications on ES5 for some time, and we may want to adapt this specific rule. I think it is overly strict. I understand the preference while chaining functions.

```javascript
promise()
  .then(function(v) {
    // does things

    return whatever;
  })
  .then(function(w) {
    // does more things

    return whatever2;
  })
```

This looks bad, and is hard to parse, pulling out these functions and giving them meaningful names clears things up.

On the other hand, I am totally fine with having an anonymous function definition in a `map` call.

```javascript
arr.map(function(item) {
  // do work here

  return whatever;
});
```
If anyone has rules in here that they think need changed, post here.

**Comment by DevShep** [comment]
This is a good time to implement https://github.com/yola/jsgreat/issues/18.

**Comment by DevShep** [comment]
Also, I would support favoring single quotes over double, without making it a hard requirement. Something like "prefer single quotes over double when starting a new file, but most importantly, maintain consistency."

**Comment by kahnjw** [comment]
Do you have a reason for that?

**Comment by DevShep** [comment]
I'm of the opinion that HTML attributes and what not should use double quotes, eg. `<input type="text">`. Single quotes allow that to happen.

Beyond that, it's a personal preference mixed with an observation that no one on the frontend team favors double over single when starting a new file. Might have missed instances, but none come to mind.

**Comment by kahnjw** [comment]
The fact that nobody favors double over single or vice versa doesn't mean we should be inconsistent.

FWIW: I actually do favor single, its just less pixels lit up on the page, but not strongly.

**Comment by DevShep** [comment]
> The fact that nobody favors double over single or vice versa doesn't mean we should be inconsistent.

So we're in agreement? I basically said everyone seems to favor single on the frontend team, and that would encourage consistency.

**Comment by kahnjw** [comment]
I guess, yeah? I thought you were proposing relaxing the rule, because it isn't a hard requirement as is. That language is different, but would not change how most people write javascript at Yola.
